homeassistant/components/glances/sensor.py

```diff
@@ -289,13 +294,16 @@ def available(self) -> bool:
             )
         return False
 
-    @property
-    def native_value(self) -> StateType:
-        """Return the state of the resources."""
-        value = self.coordinator.data[self.entity_description.type]
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_native_value()
+        super()._handle_coordinator_update()
 
-        if isinstance(value.get(self._sensor_label), dict):
-            return cast(
-                StateType, value[self._sensor_label][self.entity_description.key]
-            )
-        return cast(StateType, value[self.entity_description.key])
+    def _update_native_value(self) -> None:
+        """Update sensor native value from coordinator data."""
+        data = self.coordinator.data[self.entity_description.type]
+        if dict_val := data.get(self._sensor_label):
+            self._attr_native_value = dict_val.get(self.entity_description.key)
+        elif self.entity_description.key in data:
+            self._attr_native_value = data.get(self.entity_description.key)
```

Comment:
```suggestion
        if dict_val := data.get(self._sensor_label):
            self._attr_native_value = dict_val.get(self.entity_description.key)
        elif self.entity_description.key in data:
            self._attr_native_value = data.get(self.entity_description.key)
        self._attr_native_value = None
```
We should set value to `None` if key isn't found so the state goes `unknown`

Reply:
Sure I can add this.
Right now the computed uptime is not set in the data structure if it does not change, I'll change that too.

Reply:
Should be good now.